modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/RichText/RichText.es.js

```diff
@@ -227,20 +244,38 @@ const RichText = ({
 	}
 
 	const resetTranslation = useCallback(() => {
-		editorRef.current.editor.setData(currentValue[defaultLocale.localeId]);
-	}, [editorRef, currentValue, defaultLocale]);
+		const data = currentValue[defaultLocale.localeId];
+
+		if (Liferay.FeatureFlags['LPD-11235']) {
+			setCKEditor5Config({
+				...ckEditor5Config,
+				initialData: data ?? '',
+			});
+		}
+		else {
+			editorRef.current.editor.setData(data);
+		}
+	}, [ckEditor5Config, currentValue, defaultLocale, editorRef]);
 
 	useEffect(() => {
 		const handleRestoreState = () => {
-			editorRef.current.editor.setData(value);
+			if (Liferay.FeatureFlags['LPD-11235']) {
+				setCKEditor5Config({
+					...ckEditor5Config,
+					initialData: value,
+				});
+			}
+			else {
+				editorRef.current.editor.setData(value);
+			}
```

Comment:
I don't know what scenario triggers this code, it's handling of an external `ddm:restoreState` event. But the new code _should_ work, unlike existing code that would surely break,